### PR TITLE
 authn: add Audiences to TokenReview

### PR DESCRIPTION
**Pull request by mikedanese**
ref #58790

```release-note
TokenReview now accepts an Audience to allow clients to validate tokens not destined for the kube-apiserver.
```

### CODE REVIEW

File: pkg/apis/authentication/types.go [hunk 1]
```diff
@@ -59,6 +59,10 @@ type TokenReview struct {
 type TokenReviewSpec struct {
 	// Token is the opaque bearer token.
 	Token string
+	// Audiences is a list of the identifiers that the resource server presented
+	// with the token identifies as. Token authenticators will verify that the
```
Comment (liggitt) [inline pkg/apis/authentication/types.go]: "Audience-aware token authenticators"

**Comment by liggitt** [comment]
@kubernetes/sig-auth-api-reviews

**Comment by mikedanese** [comment]
Folded into https://github.com/kubernetes/kubernetes/pull/62692

**Pull request closed without merging**